src/status_im2/navigation/core.cljs

```diff
@@ -93,131 +115,89 @@
-      (set-view-id view-id)
-      (when-not @state/curr-modal
-        (reset! state/pushed-screen-id view-id)))))
-
-;; SCREEN DID DISAPPEAR
-(defn component-did-disappear-listener
-  [_]
-  #_(when-not (#{:popover :bottom-sheet :signing-sheet :visibility-status-popover :wallet-connect-sheet
-                 :wallet-connect-success-sheet :wallet-connect-app-management-sheet}
-               view-id)
-      ;; TODO what to do ?
-      (doseq [[_ {:keys [ref value]}] @quo.text-input/text-input-refs]
-        (.setNativeProps ^js ref (clj->js {:text value})))
-      (doseq [[^js text-input default-value] @react/text-input-refs]
-        (.setNativeProps text-input (clj->js {:text default-value})))))
+(re-frame/reg-fx :show-toasts
+                 (fn []
+                   (show-overlay "toasts"
+                                 {:overlay {:interceptTouchOutside false}
+                                  :layout  {:componentBackgroundColor :transparent}})))
+(re-frame/reg-fx :hide-toasts (fn [] (dissmiss-overlay "toasts")))
 
-;; APP LAUNCHED
-(defn app-launched-listener
-  []
-  (reset! state/curr-modal false)
-  (reset! state/dissmissing false)
-  (if (= @state/root-id :multiaccounts-stack)
-    (re-frame/dispatch-sync [::set-multiaccount-root])
-    (when @state/root-id
-      (navigation/set-root (get (roots/roots) @state/root-id))
-      (re-frame/dispatch [::login-core/check-last-chat])))
-  (rn/hide-splash-screen))
-
-;; SET ROOT
-(re-frame/reg-fx
- :init-root-fx
- (fn [new-root-id]
-   (let [root (get (roots/roots) new-root-id)]
-     (log/debug :init-root-fx new-root-id)
-     (dismiss-all-modals)
-     (re-frame/dispatch [:multiaccounts.ui/switch-theme
-                         (get roots/themes new-root-id)
-                         new-root-id])
-     (reset! state/root-id (or (get-in root [:root :stack :id]) new-root-id))
-     (navigation/set-root root))))
+;; bottom sheet
+(navigation/register-component "bottom-sheet"
+                               (fn [] (gesture/gesture-handler-root-hoc views/bottom-sheet))
+                               (fn [] views/bottom-sheet))
 
-(rf/defn set-multiaccount-root
-  {:events [::set-multiaccount-root]}
-  [{:keys [db]}]
-  (log/debug :set-multiaccounts-root)
-  (let [key-uid          (get-in db [:multiaccounts/login :key-uid])
-        keycard-account? (boolean (get-in db
-                                          [:multiaccounts/multiaccounts
-                                           key-uid
-                                           :keycard-pairing]))]
-    {:init-root-fx (if keycard-account? :multiaccounts-keycard :multiaccounts)}))
+(re-frame/reg-fx :show-bottom-sheet (fn [] (show-overlay "bottom-sheet")))
+(re-frame/reg-fx :hide-bottom-sheet (fn [] (dissmiss-overlay "bottom-sheet")))
+
+;; MERGE OPTIONS
+(re-frame/reg-fx
+ :merge-options
+ (fn [{:keys [id options]}]
+   (navigation/merge-options id options)))
 
+;; LEGACY (should be removed in status 2.0)
 (defn get-screen-component
```

Comment:
how about we just rename the function `get-screen-component-DEPRECATED` or something to this effect? it's a bit more noticeable that we need to remove it then

Reply:
Just to add to this idea, we can also add the metadata `^:deprecated` to any var and `clj-kondo` can be configured to warn (instead of fail). Some editors also strikethrough usages of deprecated vars.

Reply:
this will be removed with old code, we still merge old screens , so i could just move this into status-im namespace, no need to maintain or care about this code